netweaver/monitoring.sls

```diff
@@ -1,26 +1,36 @@
 {%- from "netweaver/map.jinja" import netweaver with context -%}
 {% set host = grains['host'] %}
 
+
 prometheus_sap_host_exporter_pkg:
   pkg.installed:
     - name: prometheus-sap_host_exporter
 
+# the sid, instance number pair of a node is unique, so we need to adapt configuration
+{% for node in netweaver.nodes if host == node.host %}
+
+{% set instance = node.instance %}
+{% set instance = '{:0>2}'.format(node.instance) %}
+{% set daemon_instance = '{}_{}'.format(node.sid, instance) %}
+
+# don't install exporter for db instance
+{%- if not node.sap_instance == "db" %}
+
 sap_host_exporter_exporter_service:
   service.running:
-    - name: prometheus-sap_host_exporter
+    - name: prometheus-sap_host_exporter@{{ daemon_instance }}
     - enable: True
     - restart: True
     - require:
       - pkg: prometheus_sap_host_exporter_pkg
-      - file: sap_host_exporter_configuration
+      - file: sap_host_exporter_configuration_{{ daemon_instance }}
 
-# for each node the instance number changes so we need to adapt configuration
-{% for node in netweaver.nodes if host == node.host %}
-sap_host_exporter_configuration:
+# we can have multiple exporters in the same host in that moment, we need to change the port used by the exporter
+sap_host_exporter_configuration_{{ daemon_instance }}:
   file.managed:
-    - name: /etc/sap_host_exporter.yaml
+    - name: /etc/sap_host_exporter{{ daemon_instance }}.yaml
```

Comment:
As @stefanotorresi suggest online, I would put the config file in a folder, not just directly in `/etc`
Something like:
`/etc/sap_host_exporter/HA1-00.yaml` or `/usr/etc/sap_host_exporter/HA1-00.yaml`
For that, you will need to change the exporter spec file to create those folders

Reply:
why this? In the past we agreed already this for the `ha_cluster_exporter`. If we are going to change it, we will need to unify also ha_cluster_exporter. I mean it is something we have previously agreed on that, having the conf file without folder. I think is ok, there are other config file in `etc` from different apps.

Reply:
@MalloZup This is what I understood from @stefanotorresi when we talked on the chat. Anyway, if this is how `ha_cluster_exporter` go ahead, and put in `/etc`.
PD: `hanadb_exporter` was implemented before and we started using `/etc/hanadb_exporter` from the beginning. I don't really know where this `agreement` is coming from.

Reply:
And in fact, one of the reasons of having a folder instead of a unique file is that in this exporter, you might have several config files `sap_host_exporter_prd_00.yaml`, `sap_host_exporter_prd_01.yaml`, `sap_host_exporter_prd_02.yaml`, so haveing them all around `/etc` folders looks a bit ugly. That's my opinion anyway.

Reply:
ok make sense. We can create a folder via salt.